### PR TITLE
Upgrade dependencies for compatibility with newer polkadot version

### PR DESCRIPTION
**Pull request by conectado**
Hi, I work at Enjin and we are using [polkadot@127eb17a](https://github.com/paritytech/polkadot/commit/127eb17a25bbe2a9f2731ff11a65d7f8170f2373) as it's recommended in the [cumulus workshop](https://github.com/substrate-developer-hub/cumulus-workshop#versions-of-software) and we wanted to start doing some tests with Snowbridge but that polkadot version turned out to be incompatible, so I worked on the contents of this PR to solve this.

I upgraded the dependencies used in  artemis here using `cargo update` and mostly followed the changes [here](https://github.com/substrate-developer-hub/substrate-parachain-template/commit/40b858149b212e493da08d80e0fc5b06a6b0b72d) to make it work and compile (The comments are taken out from there too).

I made sure that the E2E tests passed(as mentioned in the `README.md`) but for that I needed to upgrade some dependencies in the `types` and `test` directories(The Polkadot types).

I'm opening this PR in the hopes it's of any use to you and luckily we can keep in sync. Let me know if there's anything that needs fixing or if I should point to another branch for merging or anything else.

### CODE REVIEW

**Comment by musnit** [comment]
@conectado awesome thanks for this, we'll review and try merge this week

File: relayer/yarn.lock [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (philipstanislaus) [inline relayer/yarn.lock:1]: Please remove this file

File: parachain/runtime/local/src/lib.rs [hunk 2]
```diff
@@ -306,57 +322,123 @@ impl artemis_transfer::Config for Runtime {
...
+	// Native converter for Relay-chain (Parent) location; will converts to a `Relay` origin when
+	// recognised.
 	RelayChainAsNative<RelayChainOrigin, Origin>,
-	SiblingParachainAsNative<cumulus_pallet_xcm_handler::Origin, Origin>,
+	// Native converter for sibling Parachains; will convert to a `SiblingPara` origin when
+	// recognised.
+	SiblingParachainAsNative<cumulus_pallet_xcm::Origin, Origin>,
+	// Superuser converter for the Relay-chain (Parent) location. This will allow it to issue a
+	// transaction from the Root origin.
+	ParentAsSuperuser<Origin>,
```
Comment (philipstanislaus) [inline parachain/runtime/local/src/lib.rs:373]: Do we want/need this?

File: parachain/runtime/rococo/src/lib.rs [hunk 3]
```diff
@@ -308,57 +321,123 @@ impl artemis_transfer::Config for Runtime {
...
+	// Native converter for Relay-chain (Parent) location; will converts to a `Relay` origin when
+	// recognised.
 	RelayChainAsNative<RelayChainOrigin, Origin>,
-	SiblingParachainAsNative<cumulus_pallet_xcm_handler::Origin, Origin>,
+	// Native converter for sibling Parachains; will convert to a `SiblingPara` origin when
+	// recognised.
+	SiblingParachainAsNative<cumulus_pallet_xcm::Origin, Origin>,
+	// Superuser converter for the Relay-chain (Parent) location. This will allow it to issue a
+	// transaction from the Root origin.
+	ParentAsSuperuser<Origin>,
```
Comment (philipstanislaus) [inline parachain/runtime/rococo/src/lib.rs:372]: Do we want/need this?

File: parachain/runtime/snowbridge/src/lib.rs [hunk 4]
```diff
@@ -611,12 +697,14 @@ construct_runtime!(
...
-		Utility: pallet_utility::{Pallet, Call, Event, Storage} = 19,
+		XcmpQueue: cumulus_pallet_xcmp_queue::{Pallet, Call, Storage, Event<T>} = 17,
+		LocalXcm: pallet_xcm::{Pallet, Call, Event<T>, Origin} = 18,
+		CumulusXcm: cumulus_pallet_xcm::{Pallet, Origin} = 19,
+		Transfer: artemis_transfer::{Pallet, Call, Event<T>} = 20,
+		Utility: pallet_utility::{Pallet, Call, Event, Storage} = 21,
 
 		// For dev only, will be removed in production
-		Sudo: pallet_sudo::{Pallet, Call, Config<T>, Storage, Event<T>} = 20,
+		Sudo: pallet_sudo::{Pallet, Call, Config<T>, Storage, Event<T>} = 22,
```
Comment (philipstanislaus) [inline parachain/runtime/snowbridge/src/lib.rs:707]: @vgeddes is there a reason why we still have this in the snowbridge build?

**Review by philipstanislaus — COMMENTED** [review]
Thanks! 🎉

File: parachain/runtime/snowbridge/src/lib.rs [hunk 5]
```diff
@@ -308,57 +321,123 @@ impl artemis_transfer::Config for Runtime {
...
+	// Native converter for Relay-chain (Parent) location; will converts to a `Relay` origin when
+	// recognised.
 	RelayChainAsNative<RelayChainOrigin, Origin>,
-	SiblingParachainAsNative<cumulus_pallet_xcm_handler::Origin, Origin>,
+	// Native converter for sibling Parachains; will convert to a `SiblingPara` origin when
+	// recognised.
+	SiblingParachainAsNative<cumulus_pallet_xcm::Origin, Origin>,
+	// Superuser converter for the Relay-chain (Parent) location. This will allow it to issue a
+	// transaction from the Root origin.
+	ParentAsSuperuser<Origin>,
```
[the remark this answers is not in the document]
Comment (vgeddes) [reply parachain/runtime/snowbridge/src/lib.rs:372]: Yeah, this change has been expected and we need it. Its important for common-good parachains where the relay chain council is the ultimate authority. 

I've elaborated on this a bit more in https://github.com/Snowfork/polkadot-ethereum/pull/348.

File: parachain/runtime/local/src/lib.rs (hunk 2 above)
Comment (vgeddes) [reply parachain/runtime/local/src/lib.rs:373]: yeah, see my previous comment.

File: parachain/runtime/snowbridge/src/lib.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (philipstanislaus) [reply parachain/runtime/snowbridge/src/lib.rs:372]: Okay, thanks for elaborating :)

File: relayer/yarn.lock (hunk 1 above)
Comment (conectado) [reply relayer/yarn.lock]: My bad, removed [here](https://github.com/Snowfork/polkadot-ethereum/pull/413/commits/57b896d214ba8d81a53fb20a342952c511af97ea)

**Comment by vgeddes** [comment]
Thanks @conectado! I've reviewed and it looks mostly good. There are however more changes we need to make to our `transfer` pallet to make it compatible with the latest XCM API.

I'll create a new branch based off this one, and make the necessary changes.

**Comment by conectado** [comment]
> Thanks @conectado! I've reviewed and it looks mostly good. There are however more changes we need to make to our `transfer` pallet to make it compatible with the latest XCM API.
> 
> I'll create a new branch based off this one, and make the necessary changes.

Thanks!! 🙏

**Comment by vgeddes** [comment]
I've forked this branch to #417, and will continue with the necessary XCM updates there.

Thanks so much @conectado, and will make sure to properly attribute you when I merge in #417.

**Pull request closed without merging**